PlayList.java

```diff
@@ -35,32 +35,44 @@ public Track getTrack(int index) {
      *  If the list is full, does nothing and returns false.
      *  Otherwise, appends the track and returns true. */
     public boolean add(Track track) {
-        //// replace the following statement with your code
+        if (this.size >= this.maxSize) return false;
+        this.tracks[size] = track;
+        this.size++;
         return true;
     }
 
     /** Returns the data of this list, as a string. Each track appears in a separate line. */
     //// For an efficient implementation, use StringBuilder.
     public String toString() {
-        //// replace the following statement with your code
-        return "";
+        String playList = "";
+        if (size == 0) return playList;
```

Comment:
This validation is not required, this case covered in loop condition (when size 0, it will never iterate inside)
```suggestion
```